imap_processing/ena_maps/ena_maps.py

```diff
@@ -1076,6 +1051,101 @@ def __init__(
+            Spacing angle in degrees.
+        """
+        return self.sky_grid.spacing_deg
+
+    @property
+    def binning_grid_shape(self) -> tuple[int]:
+        """
+        Shape of the AzElSkyGrid.
+
+        Returns
+        -------
+        binning_grid_shape : tuple[int]
+            Shape of the AzElSkyGrid (num_az_bins, num_el_bins).
+        """
+        return self.sky_grid.grid_shape
+
+    def to_dataset(self) -> xr.Dataset:
+        """
+        Get the SkyMap data as a formatted xarray.Dataset.
+
+        Returns
+        -------
+        xr.Dataset
+            The SkyMap data as a formatted xarray Dataset with dims and coords.
+            If the SkyMap is empty, an empty xarray Dataset is returned.
+            The 1D data used for projecting is rewrapped to a 2D grid of
+            lon/lat (AKA az/el) coordinates.
+        """
+        if len(self.data_1d.data_vars) == 0:
+            # If the map is empty, return an empty xarray Dataset,
+            # with the unaltered spatial coords of the map
+            return xr.Dataset(
+                {},
+                coords={**self.spatial_coords},
+            )
+        # Rewrap each data array in the data_1d to the original 2D grid shape
+        rewrapped_data = {}
+        for key in self.data_1d.data_vars:
+            # drop pixel dim from the end, and add the spatial coords as dims
+            rewrapped_dims = [
+                dim
+                for dim in self.data_1d[key].dims
+                if dim != CoordNames.GENERIC_PIXEL.value
+            ]
+            rewrapped_dims.extend(self.spatial_coords.keys())
+            rewrapped_data[key] = xr.DataArray(
+                spatial_utils.rewrap_even_spaced_az_el_grid(
+                    self.data_1d[key].values,
+                    self.binning_grid_shape,
+                ),
+                dims=rewrapped_dims,
+            )
+            # Add the output coordinates to the rewrapped data, excluding the pixel
+            self.non_spatial_coords.update(
+                {
+                    coord: self.data_1d[key].coords[coord]
+                    for coord in self.data_1d[key].coords
+                    if coord != CoordNames.GENERIC_PIXEL.value
+                }
+            )
+        return xr.Dataset(
+            rewrapped_data,
+            coords={**self.non_spatial_coords, **self.spatial_coords},
+        )
+
+    def to_dict(self) -> dict:
+        """
+        Convert the RectangularSkyMap object to a dictionary of properties.
+
+        Returns
+        -------
+        dict
+            Dictionary containing the map properties.
+        """
+        map_properties_dict = {
+            "sky_tiling_type": "RECTANGULAR",
+            "spice_reference_frame": self.spice_reference_frame.name,
+            "spacing_deg": self.spacing_deg,
+            "values_to_push_project": (
+                self.values_to_push_project if self.values_to_push_project else []
```

Comment:
These are initialized to empty lists now, so I think you can simplify this to `"values_to_push_project": self.values_to_push_project`?